command/command.go

```diff
@@ -175,70 +177,84 @@ func (w *backgroundCmd) Start() error {
 // KeepAlive will create a goroutine to send regular SIGCONT signal to associated command
 // a single goroutine will be launched no matter how many calls are done to KeepAlive
 func (w *backgroundCmd) KeepAlive() {
+	w.Lock()
+	defer w.Unlock()
+
 	if w.DryRun() {
 		return
 	}
 
-	w.Lock()
-	defer w.Unlock()
-
 	if w.ticker != nil {
 		return
 	}
 
 	w.ticker = time.NewTicker(cmdKeepAliveTickDuration)
 
+	w.chQuit = make(chan int)
+
 	w.log.Debug("monitoring sending SIGCONT signal to process every 1s")
 
 	go func() {
 		for {
-			if w.ticker == nil {
+			if err := w.sendSIGCONTSignal(); err != nil {
 				return
 			}
+		}
+	}()
+}
 
-			<-w.safeTicker().C
```

Comment:
okay, i do think i see now why safeTicker wasn't always safe when we were shutting down the injector

Reply:
Yes, the mutex must used during the usage of the chanel.